File: 프로그래머스/lv2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int answer = n+1;
        int binaryString1Count = get1Count(n);
        while (binaryString1Count != get1Count(answer)) {
            answer++;
        }
        return answer;
    }

    int get1Count(int num) {
        String binaryString = Integer.toBinaryString(num);
        int count = 0;
        for (char c : binaryString.toCharArray()) {
            if(c == '1') count++;
        }
        return count;
    }
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def get_bin_one_count(n):
    return len(bin(n)[2:].replace('0',''))

def solution(n):
    answer = n
    count = get_bin_one_count(n)
    while True:
        answer += 1
        if count == get_bin_one_count(answer):
            return answer